auth.go

```diff
@@ -310,6 +310,81 @@ func CreateContractInvokeRequestWithMessage(
 	}
 }
 
+func verifyAuthRequest(request protocol.AuthorizationRequestMessage) error {
+	groupIDValidationMap := make(map[int][]protocol.ZeroKnowledgeProofRequest)
+
+	for _, proofRequest := range request.Body.Scope {
+		proofRequestQuery, err := unmarshalQuery(proofRequest.Query)
+		if err != nil {
+			return err
+		}
+		groupID := proofRequestQuery.GroupID
+		if groupID != 0 {
+			existingRequests := groupIDValidationMap[groupID]
+
+			// Validate that all requests in the group have the same schema, issuer, and circuit
+			for _, existingRequest := range existingRequests {
```

Comment:
for me, this check looks ineffective. We will do an unmarshal of each request for group id more then once, in case there are several requests.

Reply:
refactored